src/main.cpp

```diff
@@ -2258,32 +2258,32 @@ static DisconnectResult DisconnectBlock(const CBlock& block, CValidationState& s
     // undo transactions in reverse order
     for (int i = block.vtx.size() - 1; i >= 0; i--) {
         const CTransaction &tx = block.vtx[i];
-        uint256 const txhash = tx.GetHash();
+        uint256 const hash = tx.GetHash();
 
-        if (fUpdateAddressIndex) {
+        if (fAddressIndex && fUpdateAddressIndex) {
```

Comment:
Is this a fix?  It seems previously only the parameter was checked but not the global for fAddressIndex, which might have been false.

Reply:
No, not a fix; previously the `fUpdateAddressIndex` argument to this function was passed in as `fAddressIndex` by one caller and `false` by the other caller. But one of your earlier comments pointed out that it would be nice if readers could search for `fAddressIndex` and easily find all the affected code. So I made this change, really only so `fAddressIndex` appears here. I also changed the first caller to pass `true` instead of `fAddressIndex`, so the functionality is the same.